metron-analytics/metron-profiler-client/src/main/java/org/apache/metron/profiler/client/ProfilerClientFactory.java

```diff
@@ -0,0 +1,36 @@
+/*
+ *
+ *  Licensed to the Apache Software Foundation (ASF) under one
+ *  or more contributor license agreements.  See the NOTICE file
+ *  distributed with this work for additional information
+ *  regarding copyright ownership.  The ASF licenses this file
+ *  to you under the Apache License, Version 2.0 (the
+ *  "License"); you may not use this file except in compliance
+ *  with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+package org.apache.metron.profiler.client;
+
+import java.util.Map;
+
+/**
+ * Responsible for creating a {@link ProfilerClient}.
+ */
+public interface ProfilerClientFactory {
```

Comment:
What are you suggesting?  The "factory" in that example is called `ExtractorCreator`.  You think this should be `ProfilerClientCreator`?

Reply:
I like "Factory" better than "Creator" tbh. I'm more referring to the enum pattern that holds the list of available options for implementations that can be created.

Reply:
I can see the need for an `Extractors`-like class, when 
(1) There are many implementations for a user to choose from
(2) A user needs to define via configuration which one to use
(3) Or we want to limit the possible implementations that can be used

In this case, the user never needs to specify which implementation to use via configuration.  There is also only a single implementation that would be used in production.  I don't see where an `Extractors`-like class would help in this case. 

Do you still thing we need this? Am I missing where this might help?

Reply:
I definitely see your point. I think it's worthwhile, for the small amount of code it is, for the same reasons we leverage interfaces and factories in the first place - it provides a template and a pattern to build off of later. So if someone comes along later and does provide another implementation, there's little ambiguity - they just fit the factory pattern and add a new option.

Reply:
Ok, you sold me.  See commit 3a29c6500.

Reply:
That's actually really good, thanks!